uwblueprint/planet-read
# Story review platform/create comment

## Notion ticket link

[Create new / reply to comment](https://www.notion.so/uwblueprintexecs/Create-new-or-reply-to-comment-27fecc42fa6d4e05a91eeddee1e12245)



## Implementation description
* added a graphql mutation to create a new comment 
* created a comment request and response DTO



## Steps to test
1. make sure you have the comments table in your DB and have populated the `story_translation_contents` table (if not, then set `erase_db_and_sync = True`) 
2. go to http://localhost:5000/graphql and test mutation ([sample query](http://localhost:5000/graphql?variables=null&operationName=createComment&query=mutation%20createComment%7B%0A%20createComment(commentData%3A%7Btime%3A%20123456%2C%20userId%3A%201%2C%20content%3A%22hello%22%2C%20storyTranslationContentId%3A%202%2C%20commentIndex%3A0%2C%20resolved%3Afalse%7D)%20%7B%0A%20%20ok%0A%20%20comment%7B%0A%20%20%20%20content%0A%20%20%20%20time%0A%20%20%20%20userId%0A%20%20%20%20id%0A%20%20%20%20storyTranslationContentId%20%0A%20%20%7D%0A%7D%0A%7D ))
3. try with different values


## What should reviewers focus on?
* `backend/python/app/graphql/mutations/comment_mutation.py`
* `backend/python/app/graphql/types/comment_type.py`
* should comment time be stored as DateTime? 

## Checklist
- [ x ] My PR name is descriptive and in imperative tense
- [ x ] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [ x ] For backend changes, I have run the appropriate linters:  `docker exec -it <backend-container-id> /bin/bash -c "black . && isort --profile black ."`
- [ x ] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR
